### DISCUSSION TITLE
do x:bind property need INotifyPropertyChanged implementation

### DISCUSSION THREAD

**Discussion by heartacker** [microsoft/microsoft-ui-xaml#7604] [Q&A]
I see some doc say，x:bind can bing to any peoperty

But It can not recieve any update when property changed

**Comment by DominicMaas** [comment]
For updating the property from code, yes, you must implement `INotifyPropertyChanged`. You can use the [MVVM Toolkit](https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/introduction) to help simplify your code if you wish.

More specifically, see [Observable Object](https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/observableobject) and [Putting things together](https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/puttingthingstogether).

```csharp
public class User : ObservableObject
{
    private string name;

    public string Name
    {
        get => name;
        set => SetProperty(ref name, value);
    }
}
```
